src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -103,7 +127,45 @@
 
     }
 
+    private InstanceCredentials createCredentialsIfNotExist(Infrastructure infrastructure, Instance instance) {
```

Comment:
it looks a bit too complex to follow. Could you break it down in maybe a separate class and/or in multiple smaller methods?

Reply:
Done.